Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -377,7 +377,7 @@ void BlockFormattingContext::layout_block_level_box(Box const& box, BlockContain
 
     OwnPtr<FormattingContext> independent_formatting_context;
     if (!box.is_replaced_box() && box.has_children()) {
-        if (box.children_are_inline()) {
+        if (box.children_are_inline() && !creates_block_formatting_context(box)) {
             layout_inline_children(verify_cast<BlockContainer>(box), layout_mode, box_state.available_inner_space_or_constraints_from(available_space));
         } else {
             independent_formatting_context = create_independent_formatting_context_if_needed(m_state, box);
```

Comment:
Can we write it like this to increase clarity somewhat?

```cpp
if (auto independent_formatting_context = create_independent_formatting_context_if_needed(m_state, box)) {
    independent_formatting_context->run(box, layout_mode, box_state.available_inner_space_or_constraints_from(available_space));
} else {
    if (box.children_are_inline())
        layout_inline_children(verify_cast<BlockContainer>(box), layout_mode, box_state.available_inner_space_or_constraints_from(available_space));
    else
        layout_block_level_children(verify_cast<BlockContainer>(box), layout_mode, box_state.available_inner_space_or_constraints_from(available_space));
}
```

Reply:
yeah, you suggestion looks more clearly indeed but `create_independent_formatting_context_if_needed` will fail here https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibWeb/Layout/FormattingContext.cpp#L158 if we call it before checking if children are inline. and we can't remove this verification that children are not inline because it will make `create_independent_formatting_context_if_needed` to create `DummyFormattingContext` when it is not needed.

Reply:
hmm, will it be ok to replace `VERIFY(!child_box.children_are_inline());` with `return {}`

Reply:
Yeah that sounds like it could be a nice solution :)